### PR TITLE
[10.0][FIX]rma_analytic. Propagate analytic to other models

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Currently it is only propagated to stock moves. Too bad.

### CODE REVIEW

File: rma/models/rma_order_line.py [hunk 1]
```diff
@@ -674,3 +676,9 @@ def action_view_rma_lines(self):
             result['views'] = [(res and res.id or False, 'form')]
             result['res_id'] = rma_lines.id
         return result
+
+    @api.constrains("partner_id", "rma_id")
+    def _check_partner_id(self):
+        if self.rma_id and self.partner_id != self.rma_id.partner_id:
+            raise UserError(_(
```
Comment (MiquelRForgeFlow) [inline rma/models/rma_order_line.py]:
```suggestion
            raise ValidationError(_(
```

File: rma_analytic/tests/test_rma_analytic.py [hunk 2]
```diff
@@ -92,5 +117,135 @@ def test_analytic(cls):
...
             for move in line.move_ids:
                 cls.assertEqual(
-                    line.analytic_account_id, move.analytic_account_id,
-                    "the analytic account is not propagated")
+                    line.analytic_account_id,
+                    move.analytic_account_id,
+                    "the analytic account is not propagated",
+                )
+
+    def test_invoice_analytic(cls):
```
Comment (MiquelRForgeFlow) [inline rma_analytic/tests/test_rma_analytic.py]:
```suggestion
    def test_invoice_analytic(self):
```

File: rma_analytic/tests/test_rma_analytic.py [hunk 3]
```diff
@@ -92,5 +117,135 @@ def test_analytic(cls):
...
+                }
+            )
+        )
+        rma_line._onchange_invoice_line_id()
+        cls.assertEqual(
+            rma_line.analytic_account_id,
+            cls.inv_line_1.account_analytic_id,
+        )
+
+    def test_invoice_analytic02(cls):
```
Comment (MiquelRForgeFlow) [inline rma_analytic/tests/test_rma_analytic.py]:
```suggestion
    def test_invoice_analytic02(self):
```

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by AaronHForgeFlow** [comment]
rma_account_analytic is deprecated with those changes, do we need installable false for that one? @mreficent

**Pull request merged**